packages/manager/cypress/e2e/core/cloudpulse/timerange-verification.spec.ts

```diff
@@ -0,0 +1,519 @@
+    const targetDate = now
+      .startOf('month')
+      .plus({ days: daysOffset })
+      .set({ hour, minute });
+
+    const actualDate = targetDate.toFormat('yyyy-LL-dd HH:mm'); // Format in GMT
+    return {
+      actualDate,
+      day: targetDate.day,
+      hour: targetDate.hour,
+      minute: targetDate.minute,
+      month: targetDate.month,
+    };
+  };
+
+  it('Implement and validate the functionality of the custom date and time picker for selecting a specific date and time range', () => {
+    // Calculates start and end dates in GMT using `getDateRangeInGMT` for testing date and time ranges.
+    const {
+      actualDate: startActualDate,
+      day: startDay,
+      hour: startHour,
+      minute: startMinute,
+    } = getDateRangeInGMT(0, 12, 15);
+    const {
+      actualDate: endActualDate,
+      day: endDay,
+      hour: endHour,
+      minute: endMinute,
+    } = getDateRangeInGMT(currentDate.getDate(), 12, 15);
+
+    // Select "Custom" from the "Time Range" dropdown
+    ui.autocomplete
+      .findByLabel('Time Range')
+      .scrollIntoView()
+      .should('be.visible')
+      .type('Custom');
+
+    // Select "Custom" from the autocomplete dropdown
+    ui.autocompletePopper.findByTitle('Custom').should('be.visible').click();
+
+    // Click on "Select Start Date" input field
+    cy.findByPlaceholderText('Select Start Date').should('be.visible').click();
+
+    // Select the start date from the calendar
+    cy.findByRole('gridcell', { name: startDay.toString() })
+      .should('be.visible')
+      .click();
+
+    // Clicks the button closest to the Clock Icon, bypassing any visible state with `force: true`.
+    cy.get('[data-testid="ClockIcon"]')
+      .closest('button')
+      .click({ force: true });
+
+    // Selects the start hour, minute, and meridiem (AM/PM) in the time picker.
+
+    cy.get('[aria-label="Select hours"]')
+      .scrollIntoView({ easing: 'linear' })
+      .within(() => {
+        cy.get(`[aria-label="${startHour} hours"]`).click({ force: true });
+      });
+
+    cy.get('[aria-label="Select minutes"]')
+      .scrollIntoView({ easing: 'linear', duration: 500 })
+      .within(() => {
+        cy.get(`[aria-label="${startMinute} minutes"]`).click({ force: true });
+      });
+
+    cy.get('[aria-label="Select meridiem"]')
+      .scrollIntoView({ easing: 'linear', duration: 500 })
+      .within(() => {
+        cy.get(`[aria-label="PM"]`).click({ force: true });
+      });
+
+    // Click the "Apply" button to confirm the start date and time
+    cy.findByRole('button', { name: 'Apply' }).should('be.visible').click();
+
+    // Assert that the start date and time is correctly displayed
+    cy.findByPlaceholderText('Select Start Date')
+      .scrollIntoView({ easing: 'linear' })
+      .should('be.visible')
+      .invoke('val') // Get the current value
+      .then(cleanText) // Clean the value
+      .should('equal', `${startActualDate} PM`);
```

Comment:
Nit, just an easier way to do this

```suggestion
    cy.findByPlaceholderText('Select Start Date')
      .scrollIntoView({ easing: 'linear' })
      .should('be.visible')
      .should('have.value', `${cleanText(startActualDate)} PM`)
```

I think there are a few places this could be applied